packages/htmlbars-runtime/lib/hooks.js

```diff
@@ -422,15 +422,18 @@ export function updateLocal(env, scope, name, value) {
   @param {Environment} env
   @param {Scope} scope
   @param {Function} block
+  @param {String} blockname
 
   Corresponds to entering a shadow template that was invoked by a block helper with
   `yieldIn`.
 
-  This hook is invoked with an opaque block that will be passed along to the
-  shadow template, and inserted into the shadow template when `{{yield}}` is used.
+  This hook is invoked with an opaque block that will be passed along
+  to the shadow template, and inserted into the shadow template when
+  `{{yield}}` is used. Optionally provide a non-default block name
+  that can be targeted by `{{yield to=blockName}}`.
 */
-export function bindBlock(env, scope, block) {
-  scope.block = block;
+export function bindBlock(env, scope, block, blockName='block') {
+  scope[blockName] = block;
```

Comment:
Yes, I understood, will do.

I originally just cheaped out on an additional object instantiation, since
this is fairly hot code.
On May 9, 2015 10:48 AM, "Robert Jackson" <notifications@github.com> wrote:

> In packages/htmlbars-runtime/lib/hooks.js
> <https://github.com/tildeio/htmlbars/pull/340#discussion_r29993115>:
>
> >  */
> > -export function bindBlock(env, scope, block) {
> > -  scope.block = block;
> > +export function bindBlock(env, scope, block, blockName='block') {
> > +  scope[blockName] = block;
>
> I think that he means:
>
> scope.blocks = scope.blocks || {};
> scope.blocks[blockName] = block;
>
> So that the blocks (with potentially arbitrary names in the future) don't
> live directly off of scope.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tildeio/htmlbars/pull/340/files#r29993115>.
>